### ISSUE TITLE
"Could not fetch basic configuration from server," "password authentication failed for user "szuru""

### ISSUE THREAD

**Issue by AverytheFurry** [rr-/szurubooru#494]
Hello, I've followed the instructions in install.md, however, I'm having some trouble understanding where this error is coming from. Are you supposed to create a user (and/or database) in postgresql manually? I don't know what password I should be putting in `.env` - maybe I'm reverse proxying it through caddy incorrectly?

```
sql_1     | 2022-07-27 20:53:26.703 UTC [68] FATAL:  password authentication failed for user "szuru"
sql_1     | 2022-07-27 20:53:26.703 UTC [68] DETAIL:  Password does not match for user "szuru".
```

**Comment by AverytheFurry** [comment]
Getting this error as well

```
client_1  | 2022/07/27 21:02:31 [error] 10#10: *19 connect() failed (113: Host is unreachable) while connecting to upstream, client: 172.18.0.1, server: , request: "GET /api/info HTTP/1.1", upstream: "http://172.18.0.3:6666/info", host: "{external server url}", referrer: "{external server url}"
```

**Comment by AverytheFurry** [comment]
My docker compose does not appear to have a password or user, it appears I was definitely meant to create one, that would be tremendously useful to have in the install.md, including how for people who don't know how

**Comment by AverytheFurry** [comment]
It appears docker has it's own postgress image - of which I have no idea how to access, and I have no clue if this image is what is being used by szurubooru or if it's using my local postgres install.. Also, it looks to be significantly out of date (postgres version 11)

**Comment by neobooru** [comment]
This should only happen if you change the POSTGRES_PASSWORD after the database has been initialized (in MOUNT_SQL). An easy fix would be to either change the password back to what it previously was, or to delete the MOUNT_SQL folder (which will obviously also **remove all your post data**). If you are just installing this (and not migrating or anything) then deleting the MOUNT_SQL should be fine (since you'd have no posts) and it should solve your issue.

**Comment by AverytheFurry** [comment]
> This should only happen if you change the POSTGRES_PASSWORD after the database has been initialized (in MOUNT_SQL). An easy fix would be to either change the password back to what it previously was, or to delete the MOUNT_SQL folder (which will obviously also **remove all your post data**). If you are just installing this (and not migrating or anything) then deleting the MOUNT_SQL should be fine (since you'd have no posts) and it should solve your issue.

What is "MOUNT_SQL"? Also, I'm still unclear if I should be creating the user under the docker version of postgres or the non-docker verison (and idk how to access the docker image for postgres)

Actually, in docker_compose.yml, it appears it's expecting the outdated docker version of postgres I never installed, which I don't know how to access.. This is also the only mention of MOUNT_SQL I can find in docker-compose.yml

```
  sql:
    image: postgres:11-alpine
    restart: unless-stopped
    environment:
      POSTGRES_USER:
      POSTGRES_PASSWORD:
    volumes:
      - "${MOUNT_SQL}:/var/lib/postgresql/data"
```

**Comment by neobooru** [comment]
In your `.env` file there is a MOUNT_SQL key, by default it points to `/var/local/szurubooru/sql` but you might've changed that. Try deleting whatever folder it points to and it should work, but again this will **wipe all your szurubooru instance data**.

You don't have to touch docker-compose.yml for the default install. Docker will automatically install the correct version, so don't worry about that. It does not matter which (if any) postgres version you have on the host.

**Comment by AverytheFurry** [comment]
I've reverted any changes I made to docker_compose.yml - I only started looking there after I had trouble with the current config

My MOUNT_SQL appears to be correct:
```
# Directory to store database files
MOUNT_SQL=/var/local/szurubooru/sql
```

Do I need to manually configure / add a postgresql user? I may have initially had the wrong password (default) under `#Database credentials` in `.env`, could this be causing an issue? I've re-ran `docker-compose pull` and `docker-compose up -d` in the time since if that's relevant.

**Comment by AverytheFurry** [comment]
Here's my full .env without the password
```
# Database credentials
POSTGRES_USER=szuru
POSTGRES_PASSWORD=secure password definitely not 1234
# This shows up on the homescreen, indicating build information
BUILD_INFO=latest

# Port to expose HTTP service
# Set to 127.0.0.1:8080 if you wish to reverse-proxy the docker's port,
# otherwise the port specified here will be publicly accessible
PORT=127.0.0.1:8080

# How many waitress threads to start
# 4 is the default amount of threads. If you experience performance
# degradation with a large number of posts, increasing this may
# improve performance, since waitress is most likely clogging up with Tasks.
THREADS=1

# URL base to run szurubooru under
# See "Additional Features" section in INSTALL.md
BASE_URL=/

# Directory to store image data
MOUNT_DATA=/var/local/szurubooru/data

# Directory to store database files
MOUNT_SQL=/var/local/szurubooru/sql
```

**Comment by neobooru** [comment]
> I may have initially had the wrong password (default) under `#Database credentials` in `.env`, could this be causing an issue? I've re-ran `docker-compose pull` and `docker-compose up -d` in the time since if that's relevant.

Yeah this is the issue. Do you have any data on the szurubooru instance you want to keep? If not then just delete the `/var/local/szurubooru` folder (so including the data and sql subfolders) and run `docker-compose up -d` again to start a fresh instance (with the new password).

**Comment by AverytheFurry** [comment]
> > I may have initially had the wrong password (default) under `#Database credentials` in `.env`, could this be causing an issue? I've re-ran `docker-compose pull` and `docker-compose up -d` in the time since if that's relevant.
> 
> Yeah this is the issue. Do you have any data on the szurubooru instance you want to keep? If not then just delete the `/var/local/szurubooru` folder (so including the data and sql subfolders) and run `docker-compose up -d` again to start a fresh instance (with the new password).

Awesome, will do! Thank you so much for the help!

**Comment by AverytheFurry** [comment]
> > I may have initially had the wrong password (default) under `#Database credentials` in `.env`, could this be causing an issue? I've re-ran `docker-compose pull` and `docker-compose up -d` in the time since if that's relevant.
> 
> Yeah this is the issue. Do you have any data on the szurubooru instance you want to keep? If not then just delete the `/var/local/szurubooru` folder (so including the data and sql subfolders) and run `docker-compose up -d` again to start a fresh instance (with the new password).

`sudo rm -R /var/local/szurubooru` didn't work :( - still getting the same error message

Just a side note - I literally created a clean debian 11 install on a virtual server for this booru alone, so I don't think it's likely to be any sort of conflict

**Comment by neobooru** [comment]
Ensure that the instance is stopped before deleting the folder (using `docker-compose stop`). If that does not work then could you try cloning the repo again? This to ensure that there are no leftover changes in `docker-compose.yml`. Also do NOT change the last 10-ish lines in `server/config.yaml`

**Comment by AverytheFurry** [comment]
> Ensure that the instance is stopped before deleting the folder (using `docker-compose stop`). If that does not work then could you try cloning the repo again? This to ensure that there are no leftover changes in `docker-compose.yml`. Also do NOT change the last 10-ish lines in `server/config.yaml`

`docker-compose stop`
`sudo rm -R /var/local/szurubooru/`
`docker-compose up -d`

Same issue
```
sql_1     | 2022-07-27 23:08:05.363 UTC [29] FATAL:  password authentication failed for user "szuru"
sql_1     | 2022-07-27 23:08:05.363 UTC [29] DETAIL:  Password does not match for user "szuru".
sql_1     | 	Connection matched pg_hba.conf line 95: "host all all all md5"
```
(I think I omitted the third line in my initial post)

again, I'm still unclear on what the password in .env should even be, I just generated a secure one

**Comment by viasux** [comment]
> Ensure that the instance is stopped before deleting the folder (using `docker-compose stop`). If that does not work then could you try cloning the repo again? This to ensure that there are no leftover changes in `docker-compose.yml`. Also do NOT change the last 10-ish lines in `server/config.yaml`

I've removed everything, deleted the docker containers, etc. then cloned the repo again, configured everything normally and I'm having the same error message.

**Comment by viasux** [comment]
> Ensure that the instance is stopped before deleting the folder (using `docker-compose stop`). If that does not work then could you try cloning the repo again? This to ensure that there are no leftover changes in `docker-compose.yml`. Also do NOT change the last 10-ish lines in `server/config.yaml`

I found the issue!! My password for postgres was too long!

**Comment by viasux** [comment]
Leaving open in case devs would like to add a warning
